src/Shape/Whisker.js

```diff
@@ -0,0 +1,230 @@
+    @extends BaseClass
+    @desc Creates SVG whisker based on an array of data.
+*/
+export default class Whisker extends BaseClass {
+
+  /**
+      @memberof Whisker
+      @desc Invoked when creating a new class instance, and overrides any default parameters inherited from BaseClass.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._endpoint = accessor("endpoint", "Rect");
+    this._endpointConfig = {
+      Circle: {
+        r: accessor("r", 5)
+      },
+      Rect: {
+        height: accessor("height", 5),
+        width: accessor("width", 20)
+      }
+    };
+    this._length = accessor("length", 25);
+    this._lineConfig = {};
+    this._orient = accessor("orient", "top");
+    this._x = accessor("x", 0);
+    this._y = accessor("y", 0);
+
+  }
+
+  /**
+      @memberof Whisker
+      @desc Draws the whisker.
+      @param {Function} [*callback*]
+      @chainable
+  */
+  render(callback) {
+
+    if (this._select === void 0) {
+      this.select(select("body").append("svg")
+        .style("width", `${window.innerWidth}px`)
+        .style("height", `${window.innerHeight}px`)
+        .style("display", "block").node());
+    }
+
+    const lineData = [];
+    this._data.forEach((d, i) => {
+
+      const orient = this._orient(d, i);
+
+      let endpointX = this._x(d, i);
+      if (orient === "left") endpointX -= this._length(d, i);
+      else if (orient === "right") endpointX += this._length(d, i);
+
+      let endpointY = this._y(d, i);
+      if (orient === "top") endpointY -= this._length(d, i);
+      else if (orient === "bottom") endpointY += this._length(d, i);
+
+      lineData.push({__d3plus__: true, data: d, i, id: i, x: this._x(d, i), y: this._y(d, i)});
+      lineData.push({__d3plus__: true, data: d, i, id: i, x: endpointX, y: endpointY});
+    });
+
+    const whiskerData = [];
+    this._data.forEach((d, i) => {
+
+      const dataObj = {};
+      dataObj.__d3plus__ = true;
+      dataObj.data = d;
+      dataObj.i = i;
+      dataObj.endpoint = this._endpoint(d, i);
+      dataObj.length = this._length(d, i);
+
+      if (dataObj.endpoint === "Circle") {
+        dataObj.r = this._endpointConfig.Circle.r(d, i);
+      }
+      else if (dataObj.endpoint === "Rect") {
+        dataObj.height = this._endpointConfig.Rect.height(d, i);
+        dataObj.width = this._endpointConfig.Rect.width(d, i);
+      }
+      const orient = this._orient(d, i);
+
+      let endpointX = this._x(d, i);
+      if (orient === "left") endpointX -= this._length(d, i);
+      else if (orient === "right") endpointX += this._length(d, i);
+
+      let endpointY = this._y(d, i);
+      if (orient === "top") endpointY -= this._length(d, i);
+      else if (orient === "bottom") endpointY += this._length(d, i);
+
+      dataObj.x = endpointX;
+      dataObj.y = endpointY;
+
+      whiskerData.push(dataObj);
+
+    });
+    this._endpointShapeData = nest().key(d => d.endpoint).entries(whiskerData);
+
+    new Line()
```

Comment:
Move the rendering of this `Line` above the construction of the endpoint data, so it comes right after where `lineData` is constructed.